### DISCUSSION TITLE
Any Recommended Translator from OpenAPI Sepcification to Zod?

### DISCUSSION THREAD

**Discussion by razb-viola** [colinhacks/zod#2894] [Q&A]
I have this:

```
FirstName:
  type: string
  min: 2
  max: 10
Foo:
  type: object
  properties:
    firstName:
      $ref: '#/components/schemas/FirstName'
  required:
    - firstName
```

And I want the translator to convert this to this:

```
z.object({
  firstName: z.string().min(2).max(2)
})
```

I have tried `kubb` however I only got a result of `firstName: z.string()` without the min and max (which are critical in validations, of course).

Now obviously I can replicate the entire constraints by myself, but it will be harder to maintain.
(btw, as a plan B I tried this: `const schema = z.object<Foo>(...)` but it doesn't work.)

[a comment by an automated account is not shown]

**Comment by MatyasZERO** [comment]
You can use this: https://openapi-zod-client.vercel.app
Also as library: https://www.npmjs.com/package/openapi-zod-client
